ethcontract-mock/src/details/mod.rs

```diff
@@ -0,0 +1,911 @@
+//! Implementation details of mock node.
+
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::future::ready;
+use std::sync::{Arc, Mutex};
+
+use ethcontract::common::abi::ethereum_types::BigEndianHash;
+use ethcontract::common::abi::{Function, StateMutability, Token};
+use ethcontract::common::hash::H32;
+use ethcontract::common::{Abi, FunctionExt};
+use ethcontract::jsonrpc::serde::Serialize;
+use ethcontract::jsonrpc::serde_json::to_value;
+use ethcontract::jsonrpc::{Call, MethodCall, Params, Value};
+use ethcontract::tokens::Tokenize;
+use ethcontract::web3::types::{
+    Bytes, CallRequest, TransactionReceipt, TransactionRequest, U256, U64,
+};
+use ethcontract::web3::{helpers, Error, RequestId, Transport};
+use ethcontract::{Address, BlockNumber, H256};
+use parse::Parser;
+use sign::verify;
+
+use crate::details::transaction::TransactionResult;
+use crate::range::TimesRange;
+use crate::CallContext;
+use std::any::Any;
+
+mod default;
+mod parse;
+mod sign;
+mod transaction;
+
+/// Mock transport.
+#[derive(Clone)]
+pub struct MockTransport {
+    /// Mutable state.
+    state: Arc<Mutex<MockTransportState>>,
+}
+
+/// Internal transport state, protected by a mutex.
+struct MockTransportState {
+    /// Chain ID.
+    chain_id: u64,
+
+    /// Current gas price.
+    gas_price: u64,
+
+    /// This counter is used to keep track of prepared calls.
+    request_id: RequestId,
+
+    /// This counter is used to keep track of mined blocks.
+    block: u64,
+
+    /// This counter is used to generate contract addresses.
+    address: u64,
+
+    /// Nonce for account.
+    nonce: HashMap<Address, u64>,
+
+    /// Deployed mocked contracts.
+    contracts: HashMap<Address, Contract>,
+
+    /// Receipts for already performed transactions.
+    receipts: HashMap<H256, TransactionReceipt>,
+}
+
+impl MockTransport {
```

Comment:
Wow! You implemented a mock transport that handles all the `ethcontract` necessary RPC calls!

One thing that I worry about is that we may make use of additional RPC methods in `ethcontract` and forget to add them here. I can't think of a good way to deal with this though.